core/src/main/java/org/everit/json/schema/ConditionalSchemaValidatingVisitor.java

```diff
@@ -0,0 +1,61 @@
+package org.everit.json.schema;
+
+
+import java.util.Arrays;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConditionalSchemaValidatingVisitor extends Visitor {
```

Comment:
@zgyorffi If you want to utilize the visitor pattern, then please follow the current convention. The visitor design pattern aims to separate the traversal and the processing of the data it handles (the schema, in this case). In this context it means that in the `Visitor` superclass the following methods should also exist:

 * `visitIfSchema(Schema ifSchema)`
 * `visitThenSchema(Schema thenSchema)`
 * `visitElseSchema(Schema elseSchema)`

And the default `Visitor#visitConditionalSchema(ConditionalSchema)` method should call these in the above order. This is the separated _traversal_ step. Then you can override this 3 method in the `ConditionalSchemaValidatingVisitor` subclass to perform the validation. The class can have some mutable fields (eg. you will want to store the `ValidationException` of the `IfSchema`, if it exists), it won't hurt thread safety, because from the caller side there is only a local reference to the `ConditionalSchemaValidatingVisitor` instance.

The reason why I ask for this change is that this current implementation _seemingly_ follows the existing visitor structure, while in fact it doesn't.